src/main.zig

```diff
@@ -381,6 +383,57 @@ pub fn Deserializer(comptime ReaderType: type) type {
             };
         }
 
+        /// Deserializes an enum (stored as an integer)
+        pub fn deserializeEnum(self: *Self, comptime T: type) ReadError!T {
+            if (@typeInfo(T) != .Enum) @compileError("Expected enum type, but found type '" ++ @typeName(T) ++ "'");
+
+            const I = comptime meta.Tag(T);
+
+            // Read the value as an integer
+            var deserialized_int = try self.deserializeInt(I);
+
+            // Validate that the integer that was read is a valid member of the enum
+            var valid: bool = false;
+
+            inline for (meta.fields(T)) |enum_field| {
+                if (enum_field.value == deserialized_int) {
+                    valid = true;
+                    break;
+                }
+            }
+
+            if (!valid) {
+                return error.MismatchingFormatType;
+            }
+
+            return @intToEnum(T, deserialized_int);
+        }
+
+        /// Deserializes a tagged union
+        pub fn deserializeUnion(self: *Self, comptime T: type) ReadError!T {
+            if (@typeInfo(T) != .Union) @compileError("Expected union type, but found type '" ++ @typeName(T) ++ "'");
+
+            if (@typeInfo(T).Union.tag_type == null) @compileError("Expected tagged union type, but found type untagged '" ++ @typeName(T) ++ "'");
+
+            const TagType = comptime meta.Tag(T);
+
+            var tag = try self.deserializeEnum(TagType);
```

Comment:
`tag` can be const. We should never have to mutate its value.

Reply:
Done 👍